libminifi/src/FlowFileRecord.cpp

```diff
@@ -74,6 +76,43 @@ FlowFileRecord::FlowFileRecord(std::map<std::string, std::string> attributes, Re
 	logger_ = Logger::getLogger();
 }
 
+FlowFileRecord::FlowFileRecord(FlowFileEventRecord *event)
+: _size(0),
+  _id(_localFlowSeqNumber.load()),
+  _offset(0),
+  _penaltyExpirationMs(0),
+  _claim(NULL),
+  _isStoredToRepo(false),
+  _markedDelete(false),
+  _connection(NULL),
+  _orginalConnection(NULL)
+{
+	_entryDate = event->getFlowFileEntryDate();
+	_lineageStartDate = event->getlineageStartDate();
+	_size = event->getFileSize();
+	_offset = event->getFileOffset();
+	_lineageIdentifiers = event->getLineageIdentifiers();
+	_attributes = event->getAttributes();
+    _snapshot = false;
+    _uuidStr = event->getFlowFileUuid();
+    uuid_parse(_uuidStr.c_str(), _uuid);
+
+    if (_size > 0)
+    {
+    	_claim = new ResourceClaim();
```

Comment:
Why does claim_ need to be a pointer? You should use a unique_ptr if that is required.